File: Javascript/C1.js
let age = [20, 23, 25, 50, 39, 28, 40, 22, 30, 35];
// Hàm sắp xếp tuổi lớn nhất cạnh nhỏ nhất, lớn nhì cạnh nhỏ nhì,...
function specialSort(array) {
  // sắp xếp mảng tăng dần
  // array.sort((a, b) => a - b);
  array = bubbleSort(array);
  // Chia mảng thành 2 phần bên trái là các phần tử nhỏ nhất, bên phải là các phần tử lớn nhất
  let left = 0;
  let right = array.length - 1;
  let result = [];
  // Kết hợp 2 phần
  while (left <= right) {
    if (left === right) {
      result.push(array[left]);
    } else {
      result.push(array[right]);
      result.push(array[left]);
    }
    left++;
    right--;
  }
  return result;
}
let result = specialSort(age);
console.log(result);

function bubbleSort(arr) {
  let n = arr.length;
  for (let i = 0; i < n - 1; i++) {
      for (let j = 0; j < n - i - 1; j++) {
          if (arr[j] > arr[j + 1]) {
              let temp = arr[j];
              arr[j] = arr[j + 1];
              arr[j + 1] = temp;
          }
      }
  }
  return arr;
}

File: Java/C1.java
public class C1 {
    public static void main(String[] args) {
        int[] age = {20, 23, 25, 50, 39, 28, 40, 22, 30, 35};
        // Sắp xếp mảng tuổi lớn nhất cạnh nhỏ nhất, lớn nhì cạnh nhỏ nhì, ...
        int[] result = specialSort(age);
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i]);
            System.out.print(" ");
        }
    }
    // Hàm sắp xếp tuổi lớn nhất cạnh nhỏ nhất, lớn nhì cạnh nhỏ nhì, ...
    public static int[] specialSort(int[] array) {
        // Sắp xếp mảng tăng dần
        bubbleSort(array);
        // Tạo mảng kết quả
        int[] result = new int[array.length];
        // Chia mảng thành 2 phần
        int left = 0;
        int right = array.length - 1;
        int index = 0;
        // Kết hợp 2 phần
        while (left <= right) {
            if (left == right) {
                result[index++] = array[left];
            } else {
                result[index++] = array[right];
                result[index++] = array[left];
            }
            left++;
            right--;
        }
        return result;
    }
    public static void bubbleSort(int[] array) {
        int n = array.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - 1 - i; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }
}
